src/main/java/com/aws/greengrass/logmanager/model/CloudWatchAttemptLogInformation.java

```diff
@@ -21,9 +22,21 @@
 @Getter
 @Setter
 public class CloudWatchAttemptLogInformation {
+    protected static final Comparator<InputLogEvent> EVENT_COMPARATOR = Comparator.comparing(InputLogEvent::timestamp);
     @Builder.Default
     private List<InputLogEvent> logEvents = new ArrayList<>();
     @Builder.Default
     private Map<String, CloudWatchAttemptLogFileInformation> attemptLogFileInformationMap = new HashMap<>();
     private String componentName;
+
+    /**
+     * Get the log events in chronological order.
+     *
+     * @return sorted events
+     */
+    public List<InputLogEvent> getLogEvents() {
```

Comment:
sorting happens every time a log event is added as well as once for each upload?

Reply:
Fixed.